test/test_response_header.rb

```diff
@@ -141,4 +142,12 @@ def test_illegal_character_in_value_when_newline
 
     refute_match("X-header: First\000 line\r\nX-header: Second Lin\037e\r\n", data)
   end
+
+  def test_header_value_array
```

Comment:
Are we introducing incompatibilies with this kind of thing in a Rack 2 app? I suppose not, since a Rack 2 app working like this is sort of by definition actually a Rack 3 app?

Reply:
I doubt a Rack 2 app will ever pass an array as a header value, they'll hit the `vs.to_s.split NEWLINE` code?